corehq/apps/locations/static/locations/js/widgets.js

```diff
@@ -14,7 +14,9 @@ hqDefine("locations/js/widgets", [
         _.each($source.select2('data'), function (result) {
             const fullLengthName = result.text || result.name;
             const truncatedName = truncateLocationName(fullLengthName, $select);
-            $select.append(new Option(truncatedName, result.id));
+            var $option = new Option(truncatedName, result.id);
+            $option.setAttribute('title', result.tooltip || result.title);
```

Comment:
`title` is just an HTML attribute thing, unrelated to select2

> Also, isn't tooltip always passed through? Or there some other form where tooltip might not be set?

I honestly don't completely get this. But how it's working is - this to set the options for the primary location dropdown. The primary location dropdown is populated via the selected values of "assigned locations". On initial page load, the already "assigned locations" are populated via this [block of code](https://github.com/dimagi/commcare-hq/blob/914573698e85cf378f1946790c8f13c9202badb9/corehq/apps/locations/static/locations/js/widgets.js#L99-L109) where the "title" attribute is defined. In this case, there is no `tooltip` property passed through to result for those initial options. If a new "assigned location" is selected, then there will a `result.tooltip` defined for those newly added options. I think this is because the newly added "assigned locations" are populated via `processResults` and the `results: data.results` uses the key "tooltip" so that becomes associated with the newly selected option.

Reply:
The `tooltip` passed from the form is for only initial data (so locations that have already been saved and are being displaying upon form load). If a new location is selected from the drop down, those newly selected elements do not have a `tooltip`. They will have only the`title` property. 
![Screen Shot 2024-08-26 at 12 50 09 PM](https://github.com/user-attachments/assets/57f5d8cd-97df-4ec0-a923-985b172405c9)